### PR TITLE
guile-commonmark: init at 0.1.2; haunt: init at 0.2.4

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Motivation for this change

New packages
Also, rewrite some expressions

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/haunt/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    makeWrapper
+    pkg-config
+  ];
+  buildInputs = [
+    guile
+  ];
+  propagatedBuildInputs = [
+    guile-commonmark
+    guile-reader
+  ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/haunt/default.nix:30]: IIRC the makeWrapper should be enough and propagatedBuildInputs shouldn't be used anymore if not absolutely required.

File: pkgs/development/guile-modules/guile-cairo/default.nix [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
+}:
 
 stdenv.mkDerivation rec {
   pname = "guile-cairo";
   version = "1.11.2";
 
   src = fetchurl {
     url = "mirror://savannah/guile-cairo/${pname}-${version}.tar.gz";
-    sha256 = "0yx0844p61ljd4d3d63qrawiygiw6ks02fwv2cqx7nav5kfd8ck2";
+    hash = "sha256-YjLU3Cxb2dMxE5s7AfQ0PD4fucp4mDYaaZIGcwlBoHs=";
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-cairo/default.nix:17]: Why did the hash but not the version number change?

File: pkgs/development/guile-modules/guile-gnome/default.nix [hunk 3]
```diff
@@ -10,20 +24,35 @@ stdenv.mkDerivation rec {
...
+    gtk2
+    guile
+    guile-cairo
+    gwrap
+    libglade
+    libgnome
+    libgnomecanvas
+    libgnomeui
+    pango
+    pkg-config
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-gnome/default.nix:43]: nativeBuildInputs

File: pkgs/applications/misc/haunt/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/applications/misc/haunt/default.nix:30]: Well, I used the guix expression for haunt as a guide here.

File: pkgs/development/guile-modules/guile-gnome/default.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/development/guile-modules/guile-gnome/default.nix:43]: All of them? Or only pkg-config?

Comment (SuperSandro2000) [reply pkgs/development/guile-modules/guile-gnome/default.nix]: Only pkg-config I think.

**Pull request merged**